### ISSUE TITLE
Copypasta error in ion_reader_seek?

### ISSUE THREAD

**Issue by MFornander** [amzn/ion-c#138]
I'm not sure if ion_reader_seek works at all even on text since there are no examples or tests AFAIK and my usage trials all fail.

There is probably a bug on line 2037 though:

```    // and then we reset the underlying parser (text or binary)
    switch(preader->type) {
    case ion_type_text_reader:
        IONCHECK(_ion_reader_text_reset(preader, tid_DATAGRAM, local_end));
        break;
    case ion_type_binary_reader:
        IONCHECK(_ion_reader_text_reset(preader, tid_DATAGRAM, local_end));
        break;
```

**Comment by raganhan** [comment]
@MFornander thanks for reporting this issue. 

Are you currently blocked because of this issue? Knowing so will help us prioritize this issue.

**Comment by MFornander** [comment]
Thanks for looking at this  We are not blocked because we could not get seeking working at all. We had to work around the blocker by reading in the whole buffer at the start. Unfortunate since we had hoped to be smarter about resource use. Did I miss the examples or unit tests that shows ionc seeking? Are you guys using it?

**Comment by raganhan** [comment]
We are not using it. Given the lack of tests and examples I believe the seeker was not fully implemented. 

When we go about fixing this we need to add unit tests

**Comment by jabbottn** [comment]
@raganhan any update on this? Would be very useful if it worked.

**Comment by tgregg** [comment]
I'm looking into this to see if we can get `ion_reader_seek` working quickly. The copy/paste error is quick, but I also need to add tests to verify that the implementation works and fix any other bugs I can identify.

**Comment by tgregg** [comment]
@MFornander @jabbottn This has been fixed in #150. Please give it a try and open another issue if you find any additional issues with `ion_reader_seek`.
